service/history/execution/mutable_state_builder.go

```diff
@@ -1181,6 +1182,25 @@ func (e *mutableStateBuilder) AddWorkflowExecutionTerminatedEvent(
 	if err := e.ReplicateWorkflowExecutionTerminatedEvent(firstEventID, event); err != nil {
 		return nil, err
 	}
+
+	domainName := e.GetDomainEntry().GetInfo().Name
+
+	e.logger.Info(
+		"Workflow execution terminated.",
+		tag.WorkflowDomainName(domainName),
+		tag.WorkflowID(e.GetExecutionInfo().WorkflowID),
+		tag.WorkflowRunID(e.GetExecutionInfo().RunID),
+		tag.WorkflowTerminationReason(reason),
+	)
+
+	re := regexp.MustCompile(`[^a-zA-Z0-9]`)
```

Comment:
sorry to be annoying, but lets move this outside the function invocation, there's no need to recompile it each time. 

By convention, go programs put contstants, type declarations up the top, so perhaps around the top of the `metrics/tags.go` we can define the regex and then just use it inside the `metrics.WorkflowTerminationReasonTag` function, thefore we don't need to remember to sanitize inline like this for every use? 

The syntax outside function closure is slightly different, so either ```golang

var safeAlphaNumericStringRE = regexp.MustCompile(`[^a-zA-Z0-9]`)

// or, within a var block like:
var (
   safeAlphaNumericStringRE = regexp.MustCompile(`[^a-zA-Z0-9]`)
)
```

and that will make it easier such that we reference the already compiled regex
```
// WorkflowTerminationReasonTag reports the reason for workflow termination
func WorkflowTerminationReasonTag(value string) Tag {
        value = safeAlphaNumericStringRE.ReplaceAllString(value, "_")
	return simpleMetric{key: workflowTerminationReason, value: value}
}
```